provider/cloudflare/cloudflare.go

```diff
@@ -393,14 +389,24 @@ func (p *CloudFlareProvider) ApplyChanges(ctx context.Context, changes *plan.Cha
 	if p.CustomHostnamesConfig.Enabled {
 		for _, e := range changes.Delete {
 			for _, target := range e.Targets {
-				cloudflareChanges = append(cloudflareChanges, p.newCloudFlareChange(cloudFlareDelete, e, target, nil))
+				change, err := p.newCloudFlareChange(cloudFlareDelete, e, target, nil)
+				if err != nil {
```

Comment:
New code, error case not covered with unit tests

![Screenshot 2025-06-22 at 10 05 03](https://github.com/user-attachments/assets/b893b2e1-250d-4c87-93e2-626c955b541c)